### PR TITLE
Fix security hotspot sorting and improve test coverage

### PR DESCRIPTION
**Pull request by nquinquenel**

### CODE REVIEW

File: its/src/test/kotlin/org/sonarlint/intellij/its/tests/SecurityHotspotTabTest.kt [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+import org.sonarqube.ws.client.WsClientFactories
+import org.sonarqube.ws.client.users.CreateRequest
+import org.sonarqube.ws.client.usertokens.GenerateRequest
+import java.io.File
+import java.time.Duration
+
+
+const val SECURITY_HOTSPOT_PROJECT_KEY = "sample-java-hotspot"
+
+@DisabledIf("isCLion", disabledReason = "No security hotspots in CLion")
```
Comment (damien-urruty-sonarsource) [inline its/src/test/kotlin/org/sonarlint/intellij/its/tests/SecurityHotspotTabTest.kt:59]: I did not know this annotation, cool!

File: its/src/test/kotlin/org/sonarlint/intellij/its/tests/SecurityHotspotTabTest.kt [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        with(remoteRobot) {
+            idea {
+                toolWindow("SonarLint") {
+                    ensureOpen()
+                    tab("Security Hotspots") { select() }
+                    content("SecurityHotspotsPanel") {
+                        jLabel("Configure binding").click()
+                    }
+                }
+                dialog("Project Settings") {
```
Comment (damien-urruty-sonarsource) [inline its/src/test/kotlin/org/sonarlint/intellij/its/tests/SecurityHotspotTabTest.kt:88]: I think this part and below already exists in other tests, do you think we could factorize ?

File: src/test/java/org/sonarlint/intellij/ui/nodes/SecurityHotspotNodeTests.java [hunk 3]
```diff
@@ -33,25 +33,22 @@
...
 
 class SecurityHotspotNodeTests {
-  private LiveSecurityHotspotNode node;
 
   @Test
   void testCount() {
-    var i = createSecurityHotspot(System.currentTimeMillis(), "rule");
-    node = new LiveSecurityHotspotNode(i, false);
+    var i = createSecurityHotspot();
+    LiveSecurityHotspotNode node = new LiveSecurityHotspotNode(i, false);
```
Comment (damien-urruty-sonarsource) [inline src/test/java/org/sonarlint/intellij/ui/nodes/SecurityHotspotNodeTests.java:40]: var ?

File: src/test/java/org/sonarlint/intellij/ui/tree/IssueTreeModelBuilderTests.java [hunk 4]
```diff
@@ -128,9 +129,9 @@ private static LiveIssue mockIssuePointer(String path, int startOffset, String r
     return ip;
   }
 
-  private static ClientInputFile mockFile(String path) {
+  private static ClientInputFile mockFile(String path) throws IOException {
     var file = mock(ClientInputFile.class);
-    when(file.getPath()).thenReturn(path);
+    when(file.contents()).thenReturn(path);
```
Comment (damien-urruty-sonarsource) [inline src/test/java/org/sonarlint/intellij/ui/tree/IssueTreeModelBuilderTests.java:134]: Is this expected? You return the path as the content of the file ? Is the content important for the test anyway ?

File: src/test/java/org/sonarlint/intellij/actions/FilterSecurityHotspotActionGroupTests.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    FilterSecurityHotspotAction action = (FilterSecurityHotspotAction) actionGroup.getChildren(event)[1];
+    action.setSelected(event, true);
+    action = (FilterSecurityHotspotAction) actionGroup.getChildren(event)[2];
+    action.setSelected(event, true);
+    action = (FilterSecurityHotspotAction) actionGroup.getChildren(event)[0];
+    action.setSelected(event, true);
+
+    verify(toolWindow).filterSecurityHotspotTab(SecurityHotspotFilters.SHOW_ALL);
+    verify(toolWindow).filterSecurityHotspotTab(SecurityHotspotFilters.LOCAL_ONLY);
+    verify(toolWindow).filterSecurityHotspotTab(SecurityHotspotFilters.EXISTING_ON_SONARQUBE);
```
Comment (damien-urruty-sonarsource) [inline src/test/java/org/sonarlint/intellij/actions/FilterSecurityHotspotActionGroupTests.java:59]: How is this interpreted by Mockito ? I think the order is not checked ? I might have split this into 3 tests, but it's just me

File: src/test/java/org/sonarlint/intellij/ui/tree/IssueTreeModelBuilderTests.java (hunk 4 above)
Comment (nquinquenel) [reply src/test/java/org/sonarlint/intellij/ui/tree/IssueTreeModelBuilderTests.java]: Removed, it was not useful

File: src/test/java/org/sonarlint/intellij/actions/FilterSecurityHotspotActionGroupTests.java (hunk 5 above)
Comment (nquinquenel) [reply src/test/java/org/sonarlint/intellij/actions/FilterSecurityHotspotActionGroupTests.java]: Order does not matter, it only checks it it was called once

File: its/src/test/kotlin/org/sonarlint/intellij/its/tests/TaintVulnerabilitiesTest.kt [hunk 6]
```diff
@@ -92,39 +75,12 @@ class TaintVulnerabilitiesTest : BaseUiTest() {
...
-                        waitFor(Duration.ofSeconds(5)) { hasText("Orchestrator") }
-                        findText("Orchestrator").click()
-                    }
-                    jbTextField().text = TAINT_VULNERABILITY_PROJECT_KEY
-                    button("OK").click()
-                    // wait for binding fully established
-                    waitFor(Duration.ofSeconds(20)) { !isShowing }
-                }
+
+                ProjectBindingUtils.bindProjectToSonarQube(
```
Comment (damien-urruty-sonarsource) [inline its/src/test/kotlin/org/sonarlint/intellij/its/tests/TaintVulnerabilitiesTest.kt:79]: static import ?

File: its/src/test/kotlin/org/sonarlint/intellij/its/tests/TaintVulnerabilitiesTest.kt [hunk 7]
```diff
@@ -174,27 +125,9 @@ class TaintVulnerabilitiesTest : BaseUiTest() {
...
-                    .setCleanPackageSonarGoals()
-                    .setProperty("sonar.login", SONARLINT_USER)
-                    .setProperty("sonar.password", SONARLINT_PWD)
-            )
-
-            val generateRequest = GenerateRequest()
-            generateRequest.name = "TestUser"
-            token = adminWsClient.userTokens().generate(generateRequest).token
-        }
+            OrchestratorUtils.executeBuildWithMaven("projects/sample-java-taint-vulnerability/pom.xml", ORCHESTRATOR);
```
Comment (damien-urruty-sonarsource) [inline its/src/test/kotlin/org/sonarlint/intellij/its/tests/TaintVulnerabilitiesTest.kt:128]: static import ?

**Pull request merged**